control/config.py

```diff
@@ -7,40 +7,143 @@
 #  Authors: anita.shekar@ibm.com, sandy.kaur@ibm.com
 #
 
-import configparser
+import os
+import socket
+from pathlib import Path
+from typing import Optional
+from pydantic import (BaseModel, Extra, IPvAnyAddress, PositiveInt, Json,
+                      FilePath, conint, constr, validator, root_validator)
 
 
-class GatewayConfig:
-    """Loads and returns config file settings.
+class GatewaySubConfig(BaseModel):
```

Comment:
I really like the idea of introducing Pydantic, and runtime type checking. BTW, given the amount of type annotations, we should also enable mypy for enforcing static type checks.

I searched for hybrids of `argparse` and `pydantic` and found:
* [pydantic-argparse](https://pydantic-argparse.supimdos.com/)
* https://github.com/edornd/argdantic

Given that the above `ConfigArgParse` can be subclassed, maybe we could create a derived class from both `ConfigArgParse` and `PydanticArgparse`?

That said, I have one question: Is this validation only to enforce input params or also internal type safety? I'm asking this because for input validation we should probably do that at `argparse` level, rather than `pydantic`. If needed we could easily implement new `argparse` [`types`](https://docs.python.org/3/library/argparse.html#type) to perform more exhaustive validation (e.g.: `IPAddress`, `Port`, ...). That said, if any of the `argparse`+`pydantic` hybrids support this (by leveraging existing Pydantic models), I'd definitely go for it!